cloud/amazon/ec2_vpc_nacl_facts.py

```diff
@@ -0,0 +1,201 @@
+    returned: success
+    type: list of complex
+    contains:
+        nacl_id:
+            description: The ID of the Network Access Control List.
+            returned: always
+            type: string
+        vpc_id:
+            description: The ID of the VPC that the NACL is attached to.
+            returned: always
+            type: string
+        is_default:
+            description: True if the NACL is the default for its VPC.
+            returned: always
+            type: boolean
+        tags:
+            description: A dict of tags associated with the NACL.
+            returned: always
+            type: dict
+        subnets:
+            description: A list of subnet IDs that are associated with the NACL.
+            returned: always
+            type: list of string
+        ingress:
+            description: A list of NACL ingress rules.
+            returned: always
+            type: list of list
+        egress:
+            description: A list of NACL egress rules.
+            returned: always
+            type: list of list
+'''
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, NoCredentialsError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+# VPC-supported IANA protocol numbers
+# http://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml
+PROTOCOL_NAMES = {'-1': 'all', '1': 'icmp', '6': 'tcp', '17': 'udp'}
+
+def list_ec2_vpc_nacls(connection, module):
+
+    nacl_ids = module.params.get("nacl_ids")
+    filters = ansible_dict_to_boto3_filter_list(module.params.get("filters"))
+
+    try:
+        nacls = connection.describe_network_acls(NetworkAclIds=nacl_ids, Filters=filters)
+    except (ClientError, NoCredentialsError) as e:
+        module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+
+    # Turn the boto3 result in to ansible_friendly_snaked_names
+    snaked_nacls = []
+    for nacl in nacls['NetworkAcls']:
+        snaked_nacls.append(camel_dict_to_snake_dict(nacl))
+
+    # Turn the boto3 result in to ansible friendly tag dictionary
+    for nacl in snaked_nacls:
+        if 'tags' in nacl:
+            nacl['tags'] = boto3_tag_list_to_ansible_dict(nacl['tags'])
+        if 'entries' in nacl:
+            nacl['egress'] = [nacl_entry_to_list(e) for e in nacl['entries']
+                              if e['rule_number'] != 32767 and e['egress']]
+            nacl['ingress'] = [nacl_entry_to_list(e) for e in nacl['entries']
+                               if e['rule_number'] != 32767 and not e['egress']]
+            del nacl['entries']
+        if 'associations' in nacl:
+            nacl['subnets'] = [a['subnet_id'] for a in nacl['associations']]
+            del nacl['associations']
+        if 'network_acl_id' in nacl:
+            nacl['nacl_id'] = nacl['network_acl_id']
+            del nacl['network_acl_id']
+
+    module.exit_json(nacls=snaked_nacls)
```

Comment:
The existing Ansible module is `ec2_vpc_nacl`, so I went with `nacl` rather than `network_acl`. The existing module also returns `nacl_id` for newly created items.